torch/distributed/distributed_c10d.py

```diff
@@ -4515,12 +4515,20 @@ def new_group(
     should be created in the same order in all processes.
 
     .. warning::
-        Using multiple process groups with the ``NCCL`` backend concurrently
-        is not safe and the user should perform explicit synchronization in
-        their application to ensure only one process group is used at a time.
-        This means collectives from one process group should have completed
-        execution on the device (not just enqueued since CUDA execution is
-        async) before collectives from another process group are enqueued.
+        When using multiple process groups with the ``NCCL`` backend, the user
+        must ensure a globally consistent execution order of collectives across
+        ranks.
+
+        If multiple threads within a process issue collectives, explicit
+        synchronization is necessary to ensure consistent ordering.
```

Comment:
How about:
If multiple threads within a process issue collectives to different process groups, explicit synchronization between threads is necessary to ensure consistent ordering.

Reply:
ok- so its actually fine for multiple threads in the process to issue collectives to one PG without doing synchronization? i thought that would also be wrong